### PR TITLE
Make `task news` nag configurable and deterministic

### LINKED ISSUE #3497: Add verbosity option to silence `task news` message
There is a message on some fraction of reports that says
```
Please run 'task news' to read highlights about the new release.
```

Let's add another `verbose` term (along with affected,blank,context,edit,header,footnote,label,new-id,project,special,sync,override,recur) to silence this particular message.

Branched from #3196.

### PR DESCRIPTION
**Pull request by Necior**
This patch fixes #3497.

### CODE REVIEW

**Comment by djmitche** [comment]
I just landed a fix for #432 which is what is causing the pre-commit.ci to fail.

**Review by djmitche — APPROVED** [review]

**Comment by Necior** [comment]
# The problem with this patch

This patch, in its current form, would disable the `task news` nag for every user whose `.taskrc` file specifies `verbose=` value as a list of flags. That's because without updating their `.taskrc` file, they would not have `news-nag` flag set.

That's why I'm considering adding a new config option instead (and hence the draft status).

# The general problem

I'm not sure whether adding new verbosity flags is a good idea in general.

Let's consider that a user has the following line in their `.taskrc`:

```
# Default value, except `new-id` got removed.
verbose=affected,blank,context,edit,header,footnote,label,project,special,sync,override,recur
```

So far so good. Now let's say they upgraded to a new Taskwarrior version that introduced a new verbosity flag, `newflag`.

If `newflag` is by default toggled off, Taskwarrior works as expected.

What happens when `newflag` is by default toggled on? Taskwarrior reads user's `.taskrc`, sees that `verbose=` has user-provided value and this value doesn't include `newflag`.

Therefore, the user never sees `newflag`-related messages even though the flag is by default toggled on and the user never explicitly toggled it off.

I'm of the opinion that each verbosity flag should have 3 possible values: Taskwarrior's default, explicitly on, explicitly off.

---

What do you think? Is this reasoning sounds or am I missing something?

**Comment by tbabej** [comment]
There might even be an issue for it, but in general I agree this design of the verbose field right now is not very upgrade friendly, exactly for the reason @Necior mentions.

IMHO the config should be changed to a trimodal value, as proposed above:
```
verbose.affected=yes   # turns on this verbosity flag
verbose.header=no      # turns off this verbosity flag
verbose.label=default  # explicit default, also the value that is set if there is no explicit configuration
```

File: doc/man/taskrc.5.in [hunk 1]
```diff
@@ -303,6 +303,7 @@ control specific occasions when output is generated. This list may contain:
     label      Column labels on tabular reports
     new-id     Provides feedback on any new task with IDs (and UUIDs for new tasks with ID 0, such as new completed tasks).
     new-uuid   Provides feedback on any new task with UUIDs. Overrides new-id. Useful for automation.
+    news-nag   Reminds to read new release highlights until the user runs "task news".
```
Comment (tbabej) [inline doc/man/taskrc.5.in]: Maybe we should simply call this `news`?

**Comment by djmitche** [comment]
Let's get an issue in to update the verbosity support, then. But, I think this can land as-is, with the `-nag` removed.

File: doc/man/taskrc.5.in (hunk 1 above)
Comment (Necior) [reply doc/man/taskrc.5.in]: Done.

**Review by djmitche — APPROVED** [review]

**Comment by Necior** [comment]
@tbabej, I searched through the backlog and found #2454 -- I suppose that's what you are referring to. Takeaway number one: I should've read the backlog before. Takeaway number two: great minds think alike ;)

**Pull request merged**